components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java

```diff
@@ -391,52 +391,63 @@ public void updateTenant(TenantInfoBean tenantInfoBean) throws Exception {
 
         tenant.setAdminFirstName(tenantInfoBean.getFirstname());
         tenant.setAdminLastName(tenantInfoBean.getLastname());
-        TenantMgtUtil.addClaimsToUserStoreManager(tenant);
-
-        // filling the email value
-        if (tenantInfoBean.getEmail() != null && !tenantInfoBean.getEmail().equals("")) {
-            // validate the email
-            try {
-                CommonUtil.validateEmail(tenantInfoBean.getEmail());
-            } catch (Exception e) {
-                String msg = "Invalid email is provided.";
-                log.error(msg, e);
-                throw new Exception(msg, e);
-            }
-            tenant.setEmail(tenantInfoBean.getEmail());
-        }
 
-        UserRealm userRealm = configSystemRegistry.getUserRealm();
         try {
-            userStoreManager = userRealm.getUserStoreManager();
-        } catch (UserStoreException e) {
-            String msg = "Error in getting the user store manager for tenant, tenant domain: " +
-                         tenantDomain + ".";
-            log.error(msg, e);
-            throw new Exception(msg, e);
-        }
+            PrivilegedCarbonContext.startTenantFlow();
+
+            PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+            carbonContext.setTenantDomain(tenantDomain);
+            carbonContext.setTenantId(tenantId);
+
+            TenantMgtUtil.addClaimsToUserStoreManager(tenant);
+
+            // filling the email value
+            if (tenantInfoBean.getEmail() != null && !tenantInfoBean.getEmail().equals("")) {
+                // validate the email
+                try {
+                    CommonUtil.validateEmail(tenantInfoBean.getEmail());
+                } catch (Exception e) {
+                    String msg = "Invalid email is provided.";
+                    log.error(msg, e);
+                    throw new Exception(msg, e);
```

Comment:
This code segment was already there. Due to formatting it is shown as a new addition.